core/src/main/java/com/opensymphony/xwork2/ognl/OgnlUtil.java

```diff
@@ -143,12 +143,18 @@ protected void setEnableExpressionCache(String cache) {
         enableExpressionCache = BooleanUtils.toBoolean(cache);
     }
 
-    @Inject(value = StrutsConstants.STRUTS_OGNL_EXPRESSION_CACHE_MAXSIZE, required = false)
```

Comment:
Doesn't make much sense to inject this both in the factory and here. Plus, Caffeine doesn't support changing the max size after construction (and neither does LRU really), so we'd have to rebuild the cache to continue supporting this method. I can't see much use for it anyway so I've just gone ahead and deprecated it.

Reply:
```java
cache.policy().eviction().ifPresent(eviction -> {
  eviction.setMaximum(2 * eviction.getMaximum());
});
```

The [Policy](https://www.javadoc.io/doc/com.github.ben-manes.caffeine/caffeine/latest/com.github.benmanes.caffeine/com/github/benmanes/caffeine/cache/Policy.html) api contains various methods for pragmatic needs, while keeping the core apis simple and focused on the common needs.